src/main/java/com/meetup/server/auth/application/CustomOAuth2UserService.java

```diff
@@ -50,7 +51,15 @@ public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2Authentic
     private User getOrGenerateUser(OAuthAttributes attributes) {
         log.info("attributes: {}", attributes);
 
-        return userRepository.findBySocialId(attributes.oauth2UserInfo().getSocialId())
+        Optional<User> user = userRepository.findBySocialIdAndDeletedAtIsNotNull(attributes.oauth2UserInfo().getSocialId());
+
+        if (user.isPresent()) {
+            User rejoinUser = user.get();
+            rejoinUser.updateUser(attributes.toEntity(attributes.oauth2UserInfo()));
+            return userRepository.save(rejoinUser);
+        }
+
+        return userRepository.findBySocialIdAndDeletedAtIsNull(attributes.oauth2UserInfo().getSocialId())
                 .orElseGet(() -> userRepository.save(attributes.toEntity(attributes.oauth2UserInfo())));
     }
```

Comment:
```suggestion
private User getOrGenerateUser(OAuthAttributes attributes) {
    String socialId = attributes.oauth2UserInfo().getSocialId();
    log.info("attributes: {}", attributes);

    return userRepository.findBySocialId(socialId)
            .map(user -> {
                if (user.getdeletedAt != null) {
                    user.updateUser(attributes.toEntity(attributes.oauth2UserInfo()));
                }
                return userRepository.save(user);
            })
            .orElseGet(() -> userRepository.save(attributes.toEntity(attributes.oauth2UserInfo())));
}
```

P2) 조회 쿼리를 2번 날리고 있는데, 1회 조회로 처리 가능할 것 같습니다!

Reply:
좋아요! 쿼리 효율까지 고려해주셔서 감사합니다!
502066e